controllers/usersController.js

```diff
@@ -278,37 +290,158 @@ UsersController.resetPassword = function(req, res){
-            
+    
                 if(databaseResult){
-                    delete databaseResult.uid; // Don't send uid to users
-                    res.status(200).send({
-                        operation: 'get',
-                        status: 'success',
-                        data: databaseResult
+                    let app = {};
+                    if(databaseResult.application){ // If user has application data in their profile
+                        app = modifyUser(databaseResult.application, data);
+                    } else { // If user has no application data in their profile (this shouldn't happen, except for test accounts)
+                        app = modifyUser(USER_SCHEMA.application, data);                   
+                    }
+    
+                    // Save modified user and update database
+                    databaseResult.application = app;
+    
+                    Database.update(COLLECTION_NAME, uid, databaseResult).then(function(){
+                        resolve({code: 200, message: "Application successfully saved", uid: uid});
+                    }).catch(function(err){
+                        reject({code: 500, message: "Application could not be saved due to: " + err});
                     });
                 } else {
-                    res.status(404).send({
-                        operation: 'get',
-                        status: 'failed',
-                        message: 'User not found'
-                    });
+                    reject({code: 404, message: 'User not found in database'});
                 }
                 
+            }).catch(function(err){
+                reject({code: 500, message: "User could not be found due to: " + err});
             });
-
+    
         }).catch(function(err){
-            res.status(404).send({
-                operation: 'get',
-                status: 'failure',
-                message: 'Token was invalid'
-            });
+            reject({code: 400, message: 'Token was invalid; Message: ' + err})
         });
-    }
+
+    });
+
+    return promise;
+}
+
+/**
+ * Saves the application data to the user's profile
+ */
+UsersController.saveApplication = function(req, res){
+
+    editApplication(req.get("authorization"), req.body).then(function(result){
+        res.sendStatus(result.code);
+    }).catch(function(err){
+        res.status(err.code).send({
+            message: err.message
+        });
+    });
+
+}
+
+/**
+ * Saves the user's application, then handles additional logic which occurs after submitting an application
+ * e.g. sending a confirmation
+ */
+UsersController.submitApplication = function(req, res){
+
+    editApplication(req.get("authorization"), req.body).then(function(result){
+        return Database.get(COLLECTION_NAME, result.uid);
+    }).then(function(dbRes){
+        return Mail.addTag(MAILING_LIST, dbRes.email, "applied");
```

Comment:
Can you change the tag to `'applied-1'`? Not sure if we'll want to differentiate later